src/expressions.md

```diff
@@ -41,28 +41,33 @@
 > &nbsp;&nbsp; &nbsp;&nbsp; | [_MatchExpression_]\
 > &nbsp;&nbsp; )
 
-An expression may have two roles: it always produces a *value*, and it may have
-*effects* (otherwise known as "side effects"). An expression *evaluates to* a
-value, and has effects during *evaluation*. Many expressions contain
-sub-expressions, called the *operands* of the expression. The meaning of each
-kind of expression dictates several things:
+Expressions are the fundamental unit of computation.
+When evaluated, they may have *effects* and evaluates to either a value or place.
+
+Where expressions are allowed are *expression contexts*.
+Expressions commonly contain sub-expressions, called the *operands* of the expression.
+
+The meaning of each kind of expression dictates several things:
 
 * Whether or not to evaluate the operands when evaluating the expression
 * The order in which to evaluate the operands
-* How to combine the operands' values to obtain the value of the expression
+* How to combine the operands' values to obtain the value or place of the expression
```

Comment:
```suggestion
* How to combine the operands' values or places to obtain the value or place of the expression
```
Expressions like `place.field` have operands that are places, so I think if we want to be consistent we need to repeat this 'or place'.

Or we could find some term for "the result of evalauting an expression", i.e., a term for "value or place". *Usually* that term is "value" but we already burned that. In the Stacked Borrows paper formalization, we hence used the term "result".